library/general/src/lib/ui/event_dispatcher.rb

```diff
@@ -37,14 +37,53 @@ def event_loop
 
       loop do
         input = user_input
-        raise "Unknown action #{input}" unless respond_to?(:"#{input}_handler")
-
-        send(:"#{input}_handler")
-
+        event_handler(input)
         return @_finish_dialog_value if @_finish_dialog_flag
       end
     end
 
+    # General dialog events handler.
+    # Can be redefined to modify the way of managing events, for example when
+    # some events need to be delegated to a widget.
+    # @example delegate events
+    #    class OKDialog
+    #     include Yast::UIShortcuts
+    #     include UI::EventDispatcher
+    #     Yast.import "UI"
+    #
+    #     def initialize
+    #       @widget = Widget.new
+    #     end
+    #
+    #     def run
+    #       return nil unless Yast::UI.OpenDialog(
+    #         HBox(
+    #           PushButton(Id(:ok), "OK"),
+    #           PushButton(Id(:cancel), "Cancel")
+    #         )
+    #       )
+    #       begin
+    #         return event_loop
+    #       ensure
+    #          Yast::UI.CloseDialog
+    #       end
+    #     end
+    #
+    #     def event_handler(input)
+    #       @widget.handler(input)
+    #       case input
+    #       when :ok
+    #         finish_dialog(:ok)
+    #       when :cancel
+    #         finish_dialog(:cancel)
+    #       end
+    #     end
+    #   end
+    def event_handler(input)
```

Comment:
@joseivanlopez well, problem your code is that it is too error prone ( you add new handler and forgot to mention it in event_handler and you already have problem. So what I suggest is to have code like
```
if respond_to?(:"#{input}_handler")
  public_send(:"#{input}_handler")
elsif respond_to?(:"handle_event")
  public_send(:"handle_event", event)
else
  raise "Unknown action #{input}"
end
```

this way

1. keep old behavior if you do not have new generic handler
2. events with handler have precedence
3. make handle_event easier code.

e.g. you example will look now like
```
def handle_event(input)
  widget.handler(input)
end
```

Consider that handle_event will be written multiple time, but code in this class only once, so better to make it smarter here and more trivial code later, which handle only its own business.